src/xbot/common/injection/wpi_factories/RealWPIFactory.java

```diff
@@ -98,14 +100,23 @@ public AdvancedJoystickButton getJoystickButton(XJoystick joystick, int button_n
     }
 
     @Override
-    public XGyro getGyro() {
+    public XGyro getGyro(ImuType imuType) {
         // It's possible that the nav6 might get disconnected, and throw some
         // exceptions
         // at runtime when it can't communicate over the serial port.
         // The robot needs to protect itself from this behavior.
         try {
-            return new Nav6Gyro();
+            switch (imuType) {
+            case nav6:
+                return new Nav6Gyro();
+            case navX:
+                return new InertialMeasurementUnitAdapter(Port.kMXP);
+                default:
+                    log.error("Could not find " + imuType.name() + "! Returning a \"broken\" MockGyro instead.");
```

Comment:
Actually, I'm not sure. This code can't be hit during tests, since it is in the RealWpiFactory, which is never directly tested due to WPILib.